### PR TITLE
optimize lsp_pos_to_pos

### PR DESCRIPTION
**Pull request by kirawi**
Seems to work.

### CODE REVIEW

File: helix-lsp/src/lib.rs [hunk 1]
```diff
@@ -66,39 +66,26 @@ pub mod util {
         pos: lsp::Position,
         offset_encoding: OffsetEncoding,
     ) -> Option<usize> {
-        let max_line = doc.lines().count().saturating_sub(1);
```
Comment (archseer) [inline helix-lsp/src/lib.rs:69]: Oh wow, that was missed in the original PR review.

File: helix-lsp/src/lib.rs [hunk 2]
```diff
@@ -66,39 +66,26 @@ pub mod util {
...
         let pos_line = pos.line as usize;
-        let pos_line = if pos_line > max_line {
+        if pos_line > doc.len_lines() - 1 {
```
Comment (archseer) [inline helix-lsp/src/lib.rs:70]: Should still use `saturating_sub` if you have `0` lines (no line endings)

**Review by archseer — CHANGES_REQUESTED** [review]

File: helix-lsp/src/lib.rs (hunk 2 above)
Comment (kirawi) [reply helix-lsp/src/lib.rs:70]: I noticed that it would return `1` even on an empty string `""`.

Comment (archseer) [reply helix-lsp/src/lib.rs:70]: Looks good then 👍🏻

**Pull request merged**